src/vs/platform/userDataProfile/node/userDataProfile.ts

```diff
@@ -30,6 +31,15 @@ export class UserDataProfilesReadonlyService extends BaseUserDataProfilesService
 		super(nativeEnvironmentService, fileService, uriIdentityService, logService);
 	}
 
+	override async init(): Promise<void> {
+		await (this.stateReadonlyService as StateReadonlyService).init();
```

Comment:
This smells like debt to me: why would this service `init` the state service? I would argue in any environment where the state service is provided to you, `init` should be called for you and you should not be calling `init`. If you need to synchronise your service on the readiness of the state service, I suggest to expose an event from the state service when it is ready to be used.

Reply:
Agreed. It was moved inside because to make the inits run in parallel based on your previous comment. I will split the initialisation of profiles into default extensions init and profiles init and init default extensions in parallel. Profiles init has to be called after state init.